stdlib/filename.mli

```diff
@@ -183,9 +183,21 @@ val get_temp_dir_name : unit -> string
 (** The name of the temporary directory:
     Under Unix, the value of the [TMPDIR] environment variable, or "/tmp"
     if the variable is not set.
-    Under Windows, the value of the [TEMP] environment variable, or "."
-    if the variable is not set.
+
+    Under Windows, it uses [GetTempPath2] (since Windows 10 21H1), or
+    [GetTempPath], and returns:
```

Comment:
```suggestion
    Under Windows:
```
(how it is implemented is an implementation detail)

Reply:
I added it because the standard libraries of Rust and Golang documented it.

Reply:
We don't do this anywhere else, so it would be a bit unusual to add it here.